lib/dealership.rb

```diff
@@ -0,0 +1,68 @@
+class Dealership
+  attr_reader :name,
+              :address,
+              :inventory
+
+  def initialize(name, address)
+    @name      = name
+    @address   = address
+    @inventory = []
+  end
+
+  def add_car(car)
+    @inventory << car
+  end
+
+  def inventory_count
+    @inventory.length
+  end
+
+  def has_inventory?
+    if inventory_count >= 1
+      true
+    else
+      false
+    end
+  end
+
+  def cars_by_make(make)
+    @inventory.select do |car|
+      car.make == make
+    end
+  end
+
+  def total_value
+    @inventory.sum do |car|
+      car.total_cost
+    end
+  end
+
+  def details
+    details = {"total_value" => total_value,
+               "address"     => @address}
+  end
+
+  def average_price_of_car
+    numerator     = total_value
+    denominator   = @inventory.length
+    average_price = (numerator / denominator)
+    result        = average_price.to_s.insert(-4, ",")
```

Comment:
Nice job figuring out how to get the comma in the string. Will this still work if we get into six digits? That is a question that I have as well.